File: 03.py
import os
import sys
import string

curr_dir = os.path.dirname(os.path.realpath(__file__))
input_file = f"{curr_dir}/03input.txt"
prio = {}
for i, ch in enumerate(list(string.ascii_lowercase) + list(string.ascii_uppercase)):
    prio[ch] = i+1

def one(file):
    with open(file, "r") as f:
        sum = 0
        for line in f:
            line = line.strip()
            half = len(line)//2
            str1 = line[:half]
            str2 = line[half:]
            for p in set(str1).intersection(str2):
                sum += prio[p]
        print(sum)

def two(file):
    with open(file, "r") as f:
        sum = 0
        group = []
        for i, line in enumerate(f):
            group.append(line.strip())
            if i%3 == 2:
                ch = set(group[0]).intersection(group[1]).intersection(group[2])
                for p in ch:
                    sum += prio[p]
                group = []
        print(sum)

one(input_file)
two(input_file)

File: 03.js
const fs = require('fs')
const readline = require('readline')
const events = require('events')
const input_file = "03input.txt"
const prio = {}
// Using for loop for (a-z):
for (let i = 1; i <= 26; i++) {
    prio[String.fromCharCode(i+96)] = i
    prio[String.fromCharCode(i+64)] = i+26
}

async function processLineByLine(f, callback){
    try {
      const rl = readline.createInterface({
        input: fs.createReadStream(f),
        crlfDelay: Infinity
      })
      rl.on('line', callback)
      await events.once(rl, 'close')
    }
    catch (err) {
      console.error(err)
    }
}

async function one(file){
    let sum = 0
    await processLineByLine(file, line => {
        let l = line.length
        let half = l / 2
        let str1 = line.substr(0, half)
        let str2 = line.substr(half, half)
        let intersect = new Set(str1.split("").filter(function(value){
            return str2.split("").includes(value)
        }))
        for (i of intersect){
            sum += prio[i]
        }
    })
    console.log(sum)
}

async function two(file){
    let sum = 0
    let group = []
    let line_nr = 0
    let intersect
    await processLineByLine(file, line => {
        group.push(line)
        if (line_nr % 3 === 2){
            intersect = new Set(group[0].split("").filter(function(value){
                return group[1].split("").includes(value) && group[2].split("").includes(value)
            }))
            for (let i of intersect){
                sum += prio[i]
            }
            group = []
        }
        line_nr++
    })
    console.log(sum)
}

async function doit(){
    await one(input_file)
    await two(input_file)
}

doit()